lib/modules/manager/regex/strategies.ts

```diff
@@ -22,7 +23,8 @@ export function handleAny(
         config
       )
     )
-    .filter(is.truthy);
+    .filter(is.truthy)
+    .filter(isValidDependency);
```

Comment:
This way the test result have not to be adapted in the same step.     
But yes, I agree it would be better to add a `skipReason` for debugging purposes. IMO this should be then changed in a second PR.